docs/providers.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Providers
+toc: true
+weight: 400
+indent: true
+---
+
+# Providers
+
+Providers are external infrastructure providers such as Amazon Web Services,
+Microsoft Azure, Google Cloud and Alibaba Cloud.
+
+In order to provision a resource in the provider, a Custom Resource Definition(CRD)
+needs to be registered in your Kubernetes cluster and its controller should
+be watching instances of that Custom Resource Definition. Provider packages
+contain many Custom Resource Definitions and their controllers to extend
+functionality of Crossplane.
+
+Here is the list of current providers:
+
+* [Amazon Web Services][provider-aws]
+  * [API Reference][aws-reference]
+* [Google Cloud][provider-gcp]
+  * [API Reference][gcp-reference]
+* [Microsoft Azure][provider-azure]
+  * [API Reference][azure-reference]
+* [Rook][provider-rook]
+  * [API Reference][rook-reference]
+* [Alibaba Cloud][provider-alibaba]
+  * [API Reference][alibaba-reference]
+
+## Installation
+
+The core Crossplane controller can install Provider controllers and CRDs for you
+through its own provider packaging mechanism, which is triggered by the application
+of a `ClusterPackageInstall` resource. For example, in order to request
+installation of the `provider-gcp` package, apply the following resource to the
+cluster where Crossplane is running:
+
+```yaml
+apiVersion: packages.crossplane.io/v1alpha1
+kind: ClusterPackageInstall
+metadata:
+  name: provider-gcp
+  namespace: crossplane-system
+spec:
+  package: "crossplane/provider-gcp:master"
+```
+
+The field `spec.package` is where you refer to the container image of the
+provider. Crossplane Package Manager will unpack that container, register
+CRDs and set up necessary RBAC rules and then start the controllers.
+
+There are a few other ways to install to trigger the installation of provider
+packages:
+
+* As part of Crossplane Helm chart by adding the following
+  statement to your `helm install` command: `--set clusterPackages.gcp.deploy=true`
+  It will install the default version hard-coded in that release of Crossplane
+  Helm chart but if you'd like to specif an exact version, you can add:
+  `--set clusterPackages.gcp.version=master`.
+* Using Crossplane kubectl plugin:
+  `kubectl crossplane package install --cluster -n crossplane-system 'crossplane/provider-gcp:master' provider-gcp`
```

Comment:
If we're going to keep this detail we should probably include a link to the crossplane-cli install docs.

Reply:
> we should probably include a link to the crossplane-cli install docs.

Updated to link CLI repo.

> If we're going to keep this detail

See my comment above. I'm open to go either way depending on what we want this page to be about.